### PR TITLE
updated 2 CDN templates for BPs

### PR DESCRIPTION
**Pull request by bmoore-msft**

### CODE REVIEW

File: 201-cdn-with-storage-account/azuredeploy.json [hunk 1]
```diff
@@ -1,76 +1,81 @@
...
     },
     {
-      "name": "[parameters('profileName')]",
+      "name": "[variables('profileName')]",
       "type": "Microsoft.Cdn/profiles",
       "location": "[resourceGroup().location]",
-      "apiVersion": "[variables('cdnApiVersion')]",
+      "apiVersion": "2015-06-01",
+      "tags": {
+        "displayName": "[variables('profileName')]"
```
Comment (philon-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: Should all resources in all templates have a displayName tag?

File: 201-cdn-with-storage-account/azuredeploy.json [hunk 2]
```diff
@@ -1,76 +1,81 @@
...
-              "hostName": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')),'2015-06-15' ).primaryEndpoints.blob,'https://',''),'/','')]"
-            }
+      },
+      "resources": [
+        {
+          "apiVersion": "2015-06-01",
+          "name": "[variables('endpointName')]",
+          "type": "endpoints",
+          "dependsOn": [
+            "[variables('profileName')]",
```
Comment (philon-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: Should we use resourceId in these dependsOn entries for consistency?

File: 201-cdn-with-storage-account/azuredeploy.json [hunk 3]
```diff
@@ -1,76 +1,81 @@
...
-        ]
-      }
+        }
+      ]
+    }
+  ],
+  "outputs": {
+    "hostName": {
+      "type": "string",
+      "value": "[reference(resourceId('Microsoft.cdn/profiles/endpoints', variables('profileName'), variables('endpointName'))).hostName]"
```
Comment (philon-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: You should be able to simplify these to: "[reference(variables('endpointName')).hostName]"

File: 201-cdn-with-web-app/azuredeploy.json [hunk 4]
```diff
@@ -1,98 +1,100 @@
...
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('serverFarmName'))]"
       }
     },
     {
-      "name": "[parameters('profileName')]",
+      "name": "[variables('profileName')]",
       "type": "Microsoft.Cdn/profiles",
       "location": "[resourceGroup().location]",
-      "apiVersion": "[variables('cdnApiVersion')]",
+      "apiVersion": "2015-06-01",
```
Comment (philon-msft) [inline 201-cdn-with-web-app/azuredeploy.json]: Microsoft.Cdn apiVersion 2016-04-02 is available

File: 201-cdn-with-storage-account/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: not required... leaving it to authors for now

File: 201-cdn-with-storage-account/azuredeploy.json (hunk 2 above)
Comment (bmoore-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: I'd like to have some of the simpler samples show the simpler syntax...

File: 201-cdn-with-storage-account/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [inline 201-cdn-with-storage-account/azuredeploy.json]: done

File: 201-cdn-with-web-app/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [inline 201-cdn-with-web-app/azuredeploy.json]: done

**Pull request merged**